src/instruments/hcp/tc038.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+"""
+Provides support for the TC038 AC crystal oven by HC Photonics.
+"""
+
+
+# IMPORTS #####################################################################
+
+
+from instruments.units import ureg as u
+
+from instruments.abstract_instruments.instrument import Instrument
+from instruments.util_fns import assume_units
+
+import serial
+
+# CLASSES #####################################################################
+
+
+class TC038(Instrument):
+    """
+    Communication with the HCP TC038 oven.
+
+    This is the older version with an AC power supply and AC heater.
+
+    It has parity or framing errors from time to time. Handle them in your
+    application.
+    """
+
+    _registers = {
+        "temperature": "D0002",
+        "setpoint": "D0120",
+    }
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the TC038 is a crystal oven.
+
+        Example usage:
+
+        >>> import instruments as ik
+        >>> import instruments.units as u
+        >>> inst = ik.hcp.TC038.open_serial('COM10')
+        >>> inst.setpoint = 45.3
+        >>> print(inst.temperature)
+        """
+        super().__init__(*args, **kwargs)
+        self.terminator = "\r"
+        self.addr = 1
+        self._monitored_quantity = None
+        self.parity = serial.PARITY_EVEN
+
+    @property
+    def parity(self):
+        """Gets / sets the communication parity."""
+        return self._file.parity
+
+    @parity.setter
+    def parity(self, newval):
+        self._file.parity = newval
```

Comment:
I think this could be moved to `serial_communicator.py`, since it's a specific property of this instrument, but in general of the serial connection. What's your opinion @scasagrande

Reply:
Yes this should be in `serial_communicator.py`

Reply:
I'm actually a little surprised that this works, but maybe my brain is mush this morning

`self._file` is going to be a `SerialCommunicator` object since this instrument only supports serial. But we don't have a `parity` property defined there, unless its getting picked up from `IOBase`.

I would just have `self._file.parity = serial.PARITY_EVEN` in `__init__` and add the corresponding property to `SerialCommunicator`

Reply:
I updated it according to your comments.